diff --git a/configure.in b/configure.in
index 3944168..58bff6b 100644
--- a/configure.in
+++ b/configure.in
@@ -98,7 +98,7 @@ then
 	ac_sys_system=next && ac_sys_release=$4
 	MACHDEP="$ac_sys_system$ac_sys_release$ac_sys_cpu"
     else
-	ac_sys_system=`uname -s`
+	ac_sys_system=MinGW
 	if test "$ac_sys_system" = "AIX" ; then
 		ac_sys_release=`uname -v`
 	else
@@ -1526,6 +1526,8 @@ if test "$WITH_CXX" = "yes"; then
     else
 	AC_MSG_RESULT(no)
     fi
+    LDFLAGS="-no-undefined $LDFLAGS"
+    AC_SUBST([LDFLAGS])
 fi
 
 # ----------------------------------------------------------------
diff --git a/libs/pdcore/pc_util.c b/libs/pdcore/pc_util.c
index e8d8fec..f874bad 100644
--- a/libs/pdcore/pc_util.c
+++ b/libs/pdcore/pc_util.c
@@ -30,8 +30,12 @@
 #include <ieeefp.h>     /* for finite */
 #endif
 
-#if defined (isfinite)
+#if defined(isfinite)
 #define PDC_ISFINITE isfinite
+#if defined(__MINGW32__)
+/* MinGW has infinite, but we still need DWORD */
+#include <windef.h>
+#endif
 #else /* isfinite */
 
 #if defined(WIN32)
